### ISSUE TITLE
UnimplementedError (UnimplementedError: getUpdateAvailability() has not been implemented.)

### ISSUE THREAD

**Issue by paulobreim** [mateusfccp/update_available#21]
Upgrading the version of this package to the latest version broke my app that used to know with older versions.

Running the following code:

```dart
import 'package:flutter/material.dart';
import 'package:update_available/update_available.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  final updateAvailability = await getUpdateAvailability();
}
```

Will result in the following error:
`Exception has occurred. UnimplementedError (UnimplementedError: getUpdateAvailability() has not been implemented.)`

The dependencies of the project were as follows:
```yaml
dependencies:
  flutter:
    sdk: flutter
  update_available: ^2.2.0
```

By my tests, it seems that the version `^2.1.0` also has this issue, but the version `2.0.0-nullsafety` works fine.

Testing with the example provided in the package also causes the issue.

**Comment by mateusfccp** [comment]
Thanks for opening the issue! I tried the example and it's indeed throwing this error. It seems that the android plugin is not being properly registered (it seems the `registerWith` is not being called), so when calling `UpdateAvailablePlatform.insance` the default `UnimplementedUpdateAvailable` is called, which results in the issue.

I am trying to understand what's going wrong in the plugin registration and fix it.

**Comment by kuhnroyal** [comment]
I fixed it in the PR, would be great if you can release asap.

**Comment by kuhnroyal** [comment]
Please test it before tho :)

**Comment by mateusfccp** [comment]
The PR autoclosed the issue. I am going to test later today to be sure it's working, and if it is I am going to release still today.

Thanks for the help, @kuhnroyal !

**Comment by mateusfccp** [comment]
By my tests, this issue is now solved on master. I am going to close this as fixed, and I'll then release 2.2.1.
